components/date-input/src/DateInputHelper.ts

```diff
@@ -0,0 +1,53 @@
+const { DateTime } = require('luxon');
+
+// logic for validating input to Date field
+export const format = (v: string): string => {
+  const pad = (size: number, v: string): string => (
+    v.padStart(size, '0')
+  );
+  const padYear = (v: string): string => {
+    const year: number = Number(v);
+
+    return (
+      year < 50
+      ? '19' + v
+      : (
+        50 < year && year < 100
+        ? '20' + v
+        : pad(4, v)
+      )
+    );
+  };
+  const isSet = (v: any): boolean => (
+    !!(v || v === 0)
+  );
+
+  const matchWithDelimiters = /^(\d{1,4})\D?(\d{1,2})\D?(\d{1,4})$/.exec(v.trim())
+  const matchNoDelimiters = /^(\d{8})$/.exec(v.trim())
```

Comment:
Yes, because we explicitly test for this case in line 32. We need to separate out the different ways people may format their dates before we can pass it into Luxon. However I can remove the '?' quantifier in the MatchWithDelimiters regex, if that makes it clearer.